src/main.rs

```diff
@@ -39,35 +44,113 @@ async fn main() {
+                    // Report the original error for full help.
+                    err1.exit();
+                }
+            }
+        }
+    };
 
-        let path = if let Some(p) = args.next() {
-            p
-        } else if let Ok(v) = std::env::var("JS_PATH") {
-            v
-        } else {
-            bail!("No path to JS file provided: either pass it as the first argument or set the JS_PATH environment variable");
-        };
+    match args.cmd {
+        Cmd::Serve(cmd) => {
+            let code = std::fs::read_to_string(&cmd.js_path).with_context(|| {
+                format!(
+                    "Could not read Javascript file at '{}'",
+                    cmd.js_path.display()
+                )
+            })?;
 
-        std::fs::read_to_string(&path)
-            .with_context(|| format!("Could not read js file at '{}'", path))?
-    };
+            // let listen = cmd.liste
+
+            let interface = if let Some(iface) = cmd.interface {
+                iface
+            } else if let Ok(value) = std::env::var("LISTEN_IP") {
+                value
+                    .parse()
+                    .context(format!("Invalid interface in LISTEN_IP:  '{value}'"))?
+            } else {
+                std::net::Ipv4Addr::UNSPECIFIED.into()
+            };
+
+            let port = if let Some(port) = cmd.port {
+                port
+            } else if let Ok(value) = std::env::var("PORT") {
+                value
+                    .parse()
+                    .context(format!("Invalid port in PORT:  '{value}'"))?
+            } else {
+                8080
+            };
+
+            let addr: SocketAddr = (interface, port).into();
+            let config = crate::server::ServerConfig { addr };
+
+            runtime::config::CONFIG
+                .set(runtime::config::Config::default().log_level(runtime::config::LogLevel::Error))
+                .unwrap();
+
+            if cmd.watch {
+                let runner = ion_runner::WatchRunner::new(cmd.js_path.clone(), cmd.max_js_threads);
+                crate::server::run_server(config, runner).await
+            } else {
+                let runner = ion_runner::IonRunner::new_request_handler(cmd.max_js_threads, code);
+                crate::server::run_server(config, runner).await
+            }
+        }
+    }
+}
+
+/// winterjs CLI
+#[derive(clap::Parser, Debug)]
+#[clap(version)]
+struct Args {
+    #[clap(subcommand)]
+    cmd: Cmd,
+}
+
+/// Available commands.
+#[derive(clap::Subcommand, Debug)]
+enum Cmd {
+    Serve(CmdServe),
+}
+
+/// Start a WinterJS webserver serving the given JS app.
+#[derive(clap::Parser, Debug)]
+struct CmdServe {
+    /// The port to listen on.
+    #[clap(short, long, env = "WINTERJS_PORT")]
+    port: Option<u16>,
+
+    /// The interface to listen on.
+    /// Defaults to 127.0.0.1
+    #[clap(long, default_value = "127.0.0.1", env = "WINTERJS_INTERFACE")]
```

Comment:
I believe 'IP' or 'listen IP' would be the more obvious name.

Reply:
Changed to _IP